### PR TITLE
Validate cp ip only cluster creation

### PR DESCRIPTION
**Pull request by ddjjia**
*Issue #, if available:*

Modify control plane ip validation in webhook

*Description of changes:*
Only check cp ip on cluster creation

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/vsphere/reconciler/reconciler.go [hunk 1]
```diff
@@ -131,15 +131,21 @@ func (r *Reconciler) ReconcileWorkerNodes(ctx context.Context, log logr.Logger,
...
-		return controller.ResultWithReturn(), nil
+	cluster := &anywherev1.Cluster{}
+	clusterName := types.NamespacedName{Namespace: clusterSpec.Cluster.ObjectMeta.Namespace, Name: cluster.Name}
+
+	if err := r.client.Get(ctx, clusterName, cluster); err != nil && apierrors.IsNotFound(err) {
+		if err = r.ipValidator.ValidateControlPlaneIPUniqueness(clusterSpec.Cluster); err != nil {
+			clusterSpec.Cluster.Status.FailureMessage = ptr.String(err.Error())
+			log.Error(err, "Unavailable control plane IP")
+			return controller.ResultWithReturn(), nil
+		}
```
Comment (vivek-koppuru) [inline pkg/providers/vsphere/reconciler/reconciler.go:146]: You are missing the actual error case handling here if the api error is different from "NotFound"

**Review by cxbrowne1207 — APPROVED** [review]
Looking good! No complaints here.

**Pull request merged**